### PR TITLE
Remove Delete-By-Query plugin

### LINKED ISSUE #18469: Deprecate (or remove?) the Delete-By-Query plugin
Now the Delete-By-Query feature has been re implemented in core (in the reindex module actually) we must decide what to do with the plugin we created for 2.0.

I'm in favor of removing it completely for 5.0 and document it as a breaking change but one might have a different opinion.

As @nik9000 commented in https://github.com/elastic/elasticsearch/pull/18329#issuecomment-220160894 :

> Remove the delete-by-query plugin. Should we just remove it in 5.0.0, call it a breaking change, and tell people to use the version in the reindex module? Is that ok because we're kind of just moving the API? It is losing some small functionality (timeouts) but it is gaining cancel, throttle, and rethrottle.

### PR DESCRIPTION
**Pull request by tlrx**
This pull request removes the Delete-By-Query plugin and document the breaking changes. It has been replaced by the Delete-By-Query API implemented in the Reindex module.

It also replaces a wrong documentation link in the JSON REST spec.

Closes #18469

### CODE REVIEW

File: docs/plugins/api.asciidoc [hunk 1]
```diff
@@ -4,18 +4,6 @@
 API extension plugins add new functionality to Elasticsearch by adding new APIs or features, usually to do with search or mapping.
 
 [float]
-=== Core API extension plugins
```
Comment (nik9000) [inline docs/plugins/api.asciidoc]: Probably a good thing for another PR: maybe we should mention modules here just to explain that the plugin system is powerful. Or something, I dunno. Maybe just a thing for a blog post.

File: docs/reference/migration/migrate_5_0/plugins.asciidoc [hunk 2]
```diff
@@ -95,6 +95,10 @@ cloud:
 
 Cloud GCE plugin has been renamed to {plugins}/discovery-gce.html[Discovery GCE plugin].
 
+==== Delete-By-Query plugin removed
+
+The Delete-By-Query plugin has been removed in favor of a new <<docs-delete-by-query,Delete By Query API>>
+implementation in core. It now supports throttling, retries and cancellation.
```
Comment (nik9000) [inline docs/reference/migration/migrate_5_0/plugins.asciidoc]: but no longer supports timeouts. Instead use the <<docs-delete-by-query-cancel,cancel API>> to cancel deletes that run too long.

**Comment by nik9000** [comment]
I looked over the docs and left a few comments but none of them are a big deal. I just skimmed the "delete the plugin" part but it looks right. LGTM.

**Comment by s1monw** [comment]
I love the stats  - trash it!

**Comment by nik9000** [comment]
Oh! You should probably check the vagrant/bats tests if you haven't already. They install all the plugins and verify that they installed. I think you'll have to nuke a few test cases there too.

**Comment by tlrx** [comment]
> Oh! You should probably check the vagrant/bats tests if you haven't already. They install all the plugins and verify that they installed. I think you'll have to nuke a few test cases there too.

Hum, I removed the `delete-by-query` mentions in `module_and_plugin_test_cases.bash` in this PR... did I miss something else?

**Comment by nik9000** [comment]
> Hum, I removed the delete-by-query mentions in module_and_plugin_test_cases.bash in this PR... did I miss something else?

That should be good enough. I didn't see it because I was searching the page for "bats" instead of "bash". Silly me!

File: docs/plugins/delete-by-query.asciidoc [hunk 3]
```diff
@@ -1,270 +0,0 @@
-[[plugins-delete-by-query]]
-=== Delete By Query Plugin
```
Comment (clintongormley) [inline docs/plugins/delete-by-query.asciidoc]: I'd add a section with the `[[plugins-delete-by-query]]` ID to the redirects page: https://raw.githubusercontent.com/elastic/elasticsearch/master/docs/plugins/redirects.asciidoc

(but first build --all the docs to ensure that you've caught any places linking to the DBQ plugin)

Comment (tlrx) [inline docs/plugins/delete-by-query.asciidoc]: I did not know about this, thanks

**Comment by tlrx** [comment]
Thanks @nik9000 and @clintongormley !

**Pull request merged**